service/javadsl/api/src/main/scala/com/lightbend/lagom/internal/javadsl/api/ServiceReader.scala

```diff
@@ -28,13 +30,28 @@ object ServiceReader {
 
   final val DescriptorMethodName = "descriptor"
 
+  private val isJava8 = java.lang.management.ManagementFactory.getRuntimeMXBean.getSpecVersion.startsWith("1.8")
```

Comment:
Why not `System.getProperty("java.version")` or `System.getProperty("java.specification.version")`?

Reply:
So I'm torn on this.  IIUC using an MXBean you can't fake the version, which you can do trivially with system properties.  However, I'm not sure which is preferable: being able to fake or not being able to fake.

I went with this, but if someone has a strong argument I'm happy to be convinced to switch.

Reply:
I think allowing users to overwrite with system properties is actually a feature.

Reply:
@ignasi35 in which scenario is this useful? I mean, why would I override such property when running the application?

Reply:
When I use an esoteric JVM that doesn't conform with the naming or if I want to fallback to the behavior in `java8` (for whichever sneaky bug we can't foresee).

I wouldn't consider it public API, but neither a secret.

Reply:
> When I use an esoteric JVM that doesn't conform with the naming

Makes sense, but do we support such esoteric JVMs?

> if I want to fallback to the behavior in java8 (for whichever sneaky bug we can't foresee).

But then it won't work if I understand correctly. I mean, if I'm running my application using Adopt OpenJDK 11 and set `java.version` to `1.8`, then the combination of [this block](https://github.com/lagom/lagom/pull/1780/files#diff-0e432ac45a8795660faf112d8af82d76R41) and the JVM will be invalid, right?

Reply:
So I looked into it myself a bit, and RuntimeMXBean is defined in terms of the system properties and it is API (mostly?) intended for remote JVM debugging (thus JMX).  Of course, it could be used for local JVMs too.  Seeing as this is just a piece of local code I'm going to remove the indirection and just use the system property "java.vm.specification.version".